05week/spaceTravelToMars.js

```diff
@@ -10,6 +10,85 @@ let jobTypes = {
 };
 
 // Your code here
+class CrewMember {
+  constructor(name, job, specialSkill, ship){
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+    this.ship = null;
+  }
+  enterShip(ship) {
+    if(this.ship == null){
+    this.ship = ship;
+    ship.crew.push(this)
+    console.log(`${this.name} is starting on the ${this.ship.shipName}`)
+    // console.log(ship.crew)
+    // console.log(this)
+    } else  {
+      // console.log(ship.crew)
+      ship.crew.pop()
+      // console.log(ship.crew)
+      
+      ship.crew.push(this)
+      // console.log(ship.crew)
+      console.log(`${this.name} is now on board the ${ship.shipName}`)
+      // console.log(ship.name + " " + ship.crew.name)
+    
+    }
+
+  }
+}
+
+class ship{
+  constructor(shipName, type, ability, crew){
+  this.shipName = shipName;
+  this.type = type;
+  this.ability = ability;
+  this.crew = [];
+}
+
+  missionStatement() {
+    // console.log("Hello, this ship is " + this.shipName);
+    // console.log(this.crew.job)
+    // let specificJob = this.crew[0].job
+    // console.log(specificJob)
+    // console.log(jobTypes[specificJob])
+
+    // console.log(this.crew[0].job)
+    // // console.log(this.crew)
+    // console.log(this.type)
+    for(let i = 0; i < this.crew.length; i++){
+      // console.log()
+    if (jobTypes[this.crew[i].job] == this.type){
+      console.log(`The ${this.shipName}'s special ability is ${this.ability}`)
+       return `The ${this.shipName}'s special ability is ${this.ability}`;
+    
+     } else {
+        this.ability = "Can't perform a mission yet.";
+        console.log(`The ${this.shipName}'s special ability is ${this.ability}`)
+        return `The ${this.shipName}'s special ability is ${this.ability}`;
```

Comment:
you would want to loop through the entire list of crewmembers befor returning that the ship cannot perform it's mission.